scripts/src/java/org/oppia/android/scripts/proto/maven_dependencies.proto

```diff
@@ -0,0 +1,102 @@
+syntax = "proto3";
+
+package proto;
+
+option java_package = "org.oppia.android.scripts.proto";
+option java_multiple_files = true;
+
+// Represents the list of all third-party maven dependencies.
+message MavenDependencyList {
+  // Stores the list of all third-party maven dependencies.
+  repeated MavenDependency maven_dependency = 1;
+}
+
+// Stores all the information about a third-party maven dependency.
+message MavenDependency {
+  // Name of the third-party dependency.
+  string artifact_name = 2;
+
+  // Version of the third-party dependency.
+  string artifact_version = 3;
+
+  // List of licenses corresponding to a third-party dependency.
+  repeated License license = 4;
+
+  // Represents the origin of the source of the license links.
+  OriginOfLicenses origin_of_license = 5;
+}
+
+// Stores all the details corresponding to a copyright license.
+message License {
+  // Name of the license.
+  string license_name = 1;
+
+  // Primary link of the license extracted from the POM file of the dependency.
+  string primary_link = 2;
+
+  // Alternate link that would be provided manually if the primary link can't be directly
+  // extracted, or is invalid.
+  string secondary_link = 3;
+
+  // Represents whether the type of the primary_link .
+  PrimaryLinkType primary_link_type = 4;
+
+  // Represents whether the type of the secondary_link .
+  SecondaryLinkType secondary_link_type = 5;
```

Comment:
As discussed, I suggest indicating this is an alternative vs. secondary.

That being said, how does the pipeline decide when to use primary vs. the alternative link? I'm still not sure if I have a great grasp on when we would want to use the alternative link or why it's necessary.

Reply:
Done.

The pipeline will decide whether to use primary vs alternative link on the basis of `primary_link_type` - 
```kotlin
when(primary_link_type) {
  PrimaryLinkType.SCRAPE_DIRECTLY ->/ / Use primary_link
  PrimaryLinkType.SCRAPE_FROM_LOCAL_COPY -> // Use alternative link
  PrimaryLinkType.INVALID_LINK -> // Use alternative link
  PrimaryLinkType.SHOW_LINK_ONLY -> // Use primary link
}
```

Reply:
Done.

The pipeline will decide whether to use primary vs alternative link on the basis of `primary_link_type` - 
```kotlin
when(primary_link_type) {
  PrimaryLinkType.SCRAPE_DIRECTLY ->  // Use primary_link
  PrimaryLinkType.SCRAPE_FROM_LOCAL_COPY -> // Use alternative link
  PrimaryLinkType.SHOW_LINK_ONLY -> // Use primary link
  else -> Throw Exception("Human Intervention is needed to update maven_dependencies.textproto")
}
```

Reply:
Updated the pipeline as per the suggestions.